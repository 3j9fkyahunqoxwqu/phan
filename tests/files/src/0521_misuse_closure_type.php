<?php

call_user_func(function(string $arg) {
    // Phan detects invalid uses of Closures and classes.
    $i = rand(0,1);
    var_export(new $i());
    $o = new stdClass();
    var_export(new $o());
    $c = function(string $x) {};
    var_export(new $c());
    call_user_func([function () {}, 'name']);
    echo (function() {})->name();
    call_user_func([2, 'name']);
}, 'stdClass');
