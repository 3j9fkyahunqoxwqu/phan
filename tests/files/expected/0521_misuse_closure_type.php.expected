%s:6 PhanNonClassMethodCall Call to method __construct on non-class type int
%s:6 PhanTypeExpectedClassName Expected the name of a class but saw expression with type int
%s:8 PhanTypeExpectedClassName Expected the name of a class but saw expression with type \stdClass
%s:10 PhanAccessMethodPrivate Cannot access private method \Closure::__construct defined at internal:0
%s:10 PhanTypeExpectedClassName Expected the name of a class but saw expression with type Closure(string):void
%s:11 PhanUndeclaredMethodInCallable Call to undeclared method name in callable. Possible object type(s) for that method are Closure():void
%s:12 PhanUndeclaredMethod Call to undeclared method \Closure::name
%s:13 PhanTypeInvalidCallableObjectOfMethod In a place where phan was expecting a callable, saw a two-element array with a class or expression with an unexpected type 2 (expected a class type or string). Method name was name
